hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusUpdateTopicHandler.java

```diff
@@ -198,47 +231,44 @@ private void resolveMutableBuilderAttributes(
             } else {
                 builder.submitKey(newSubmitKey);
             }
-        } else {
-            builder.submitKey(topic.submitKey());
         }
         if (op.hasMemo()) {
             builder.memo(op.memo());
-        } else {
-            builder.memo(topic.memo());
         }
-        final var resolvedExpiryMeta = resolvedUpdateMetaFrom(handleContext.expiryValidator(), op, topic);
+        final var resolvedExpiryMeta = resolvedUpdateMetaFrom(handleContext.expiryValidator(), op, currentExpiryMeta);
         builder.expirationSecond(resolvedExpiryMeta.expiry());
         builder.autoRenewPeriod(resolvedExpiryMeta.autoRenewPeriod());
         if (op.hasAutoRenewAccount() && designatesAccountRemoval(op.autoRenewAccount())) {
             builder.autoRenewAccountId((AccountID) null);
         } else {
             builder.autoRenewAccountId(resolvedExpiryMeta.autoRenewAccountId());
         }
+        if (op.hasFeeScheduleKey()) {
+            builder.feeScheduleKey(op.feeScheduleKey());
+        }
+        if (op.hasFeeExemptKeyList() && !op.feeExemptKeyList().keys().isEmpty()) {
+            builder.feeExemptKeyList(op.feeExemptKeyList().keys());
+        }
+        if (op.hasCustomFees()) {
+            builder.customFees(op.customFees().fees());
+        }
     }
 
     private void validateMaybeNewAttributes(
-            @NonNull final HandleContext handleContext,
-            @NonNull final ConsensusUpdateTopicTransactionBody op,
-            @NonNull final Topic topic) {
-        validateMaybeNewAdminKey(handleContext.attributeValidator(), op);
-        validateMaybeNewSubmitKey(handleContext.attributeValidator(), op);
-        validateMaybeNewMemo(handleContext.attributeValidator(), op);
-        validateMaybeNewExpiry(handleContext.expiryValidator(), op, topic);
-    }
-
-    private void validateMaybeNewExpiry(
-            @NonNull final ExpiryValidator expiryValidator,
-            @NonNull final ConsensusUpdateTopicTransactionBody op,
-            @NonNull final Topic topic) {
-        resolvedUpdateMetaFrom(expiryValidator, op, topic);
```

Comment:
I deleted this method because the `resolvedUpdateMetaFrom` method is called twice in the `handle` - once from `validateMaybeNewAttributes.validateMaybeNewExpiry` and once from `resolveMutableBuilderAttributes`